query/src/main/java/org/infinispan/query/clustered/DistributedIterator.java

```diff
@@ -206,6 +207,8 @@ static class KeyAndScore {
       final Object key;
       final float score;
 
+      Object covertedKey;
```

Comment:
This can be removed if you use stream method I mentioned in the previous comment.

Reply:
if the solution is not applied for keeping the sorting, I don't know if we can